src/main/java/com/blackducksoftware/integration/hub/alert/channel/hipchat/HipChatChannel.java

```diff
@@ -136,29 +135,24 @@ public String testMessage(final HipChatConfigEntity config) throws IntegrationEx
     }
 
     private String createHtmlMessage(final ProjectData projectData) {
-        final StringBuilder htmlBuilder = new StringBuilder();
-        htmlBuilder.append("<strong>" + projectData.getProjectName() + " > " + projectData.getProjectVersion() + "</strong>");
-
-        final Map<NotificationCategoryEnum, CategoryData> categoryMap = projectData.getCategoryMap();
-        if (categoryMap != null) {
-            for (final NotificationCategoryEnum category : NotificationCategoryEnum.values()) {
-                final CategoryData data = categoryMap.get(category);
-                if (data != null) {
-                    htmlBuilder.append("<br />- - - - - - - - - - - - - - - - - - - -");
-                    htmlBuilder.append("<br />Type: " + data.getCategoryKey());
-                    htmlBuilder.append("<br />Number of Changes: " + data.getItemCount());
-                    for (final ItemData item : data.getItemList()) {
-                        final Map<String, Object> dataSet = item.getDataSet();
-                        htmlBuilder.append("<p>  Rule: " + dataSet.get(ItemTypeEnum.RULE.toString()));
-                        htmlBuilder.append(" | Component: " + dataSet.get(ItemTypeEnum.COMPONENT.toString()));
-                        htmlBuilder.append(" [" + dataSet.get(ItemTypeEnum.VERSION.toString()) + "]</p>");
-                    }
-                }
+        String html = "SOMETHING WENT WRONG";
```

Comment:
Would we rather throw a RuntimeException or send an error message to the channel? Obviously this message could be more in depth.

Reply:
I think freemarkerTemplatingService. getResolvedTemplate() should throw an exception if something went wrong, otherwise we would need to have the same error handling here and in EmailMessagingService